mindsdb/integrations/handlers/ollama_handler/ollama_handler.py

```diff
@@ -45,11 +34,29 @@ def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[
         args['target'] = target
         self.model_storage.json_set('args', args)
 
-        # download model
-        # TODO v2: point Ollama to the engine storage folder instead of their default location
-        model_name = args['model_name']
-        # blocking operation, finishes once model has been fully pulled and served
-        requests.post(OllamaHandler.SERVE_URL + '/api/pull', json={'name': model_name})
+        def _model_check():
+            """ Checks model has been pulled and that it works correctly. """
+            try:
+                return requests.post(
+                    OllamaHandler.SERVE_URL + '/api/generate',
+                    json={
+                        'model': args['model_name'],
+                        'prompt': 'Hello.',
+                    }
+                ).status_code
+            except Exception:
+                return 500
+
+        # check model
+        response = _model_check()
+        if response != 200:
+            # pull model (blocking operation) and serve
+            # TODO: point to the engine storage folder instead of default location
+            requests.post(OllamaHandler.SERVE_URL + '/api/pull', json={'name': args['model_name']})
+            # try one last time
+            response = _model_check()
+            if response != 200:
```

Comment:
Just to double check since I can't find it in the Ollama docs. POST request on pull endpoint returns 200 if successful or 201?

Reply:
Tried it out and can confirm this `post` and the other requests return `200` if successful. But you're right, I don't think this is documented yet.